container/lxd/client.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package lxd
+
+import (
+	"github.com/juju/errors"
+	"github.com/lxc/lxd/client"
+)
+
+// Client extends the upstream LXD client.
+type Client struct {
+	ImageServer
+}
+
+// NewClient builds and returns a Client for high-level interaction with the
+// input LXD container server.
+func NewClient(svr lxd.ContainerServer) *Client {
+	return &Client{
+		ImageServer: ImageServer{svr},
+	}
+}
+
+// UpdateServerConfig updates the server configuration with the input values.
+func (c *Client) UpdateServerConfig(cfg map[string]interface{}) error {
+	svr, eTag, err := c.GetServer()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if svr.Config == nil {
```

Comment:
Are there any concerns here about taking ownership of the map?

Would it make more sense to always create a map?

```go
if srv.Config == nil {
    srv.Config = make(map[string]interface{})
}
for k,v := range cfg {
    srv.Config[k] = v
}
```

Reply:
I'll make this change.